scripts/jerigon_zero_benchmark.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+# ------------------------------------------------------------------------------
+set -exo pipefail
+
+# Args:
+# 1 --> Output file (Not used in the current script)
+
+# Get the number of processors for parallelism
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    num_procs=$(sysctl -n hw.physicalcpu)
+else
+    num_procs=$(nproc)
+fi
+
+# Force the working directory to always be the repository root.
+REPO_ROOT=$(git rev-parse --show-toplevel)
+PROOF_OUTPUT_DIR="${REPO_ROOT}/proofs"
+BLOCK_BATCH_SIZE="${BLOCK_BATCH_SIZE:-8}"
+
+# Logging setup
+OUTPUT_LOG="jerigon_zero_output.log"
+BLOCK_OUTPUT_LOG="jerigon_zero_block_output.log"
+PROOFS_FILE_LIST="${PROOF_OUTPUT_DIR}/proof_files.json"
+
+# Ensure necessary directories exist
+mkdir -p "$PROOF_OUTPUT_DIR"
+
+# Set environment variables for parallelism and logging
+export RAYON_NUM_THREADS=$num_procs
+export TOKIO_WORKER_THREADS=$num_procs
+export RUST_MIN_STACK=33554432
+export RUST_BACKTRACE=full
+export RUST_LOG=info
+
+# Log the current date and time
+echo "$(date +"%Y-%m-%d %H:%M:%S")" &>> "$OUTPUT_LOG"
+
+# Define the blocks to process
+blocks=(100 200 300 400 500)
+
+# Function to process each block
+process_block() {
+    local block=$1
+
+    echo "Processing block: $block" &>> "$OUTPUT_LOG"
```

Comment:
I would keep the results file minimal - maybe one line with date/time of measurement, and the timings for the blocks. Would be easy that way to open file and see how the performances are changed over weeks and months.

Reply:
https://github.com/0xPolygonZero/zk_evm/pull/751#issuecomment-2439928203

this is the expected output
let me know if we need to reduce more. i think this also gives clear info along with being concise

Reply:
Line `Processing block: 1` is redundant, does not bring any new information.